core/scripts/PrecisionErrors.js

```diff
@@ -0,0 +1,1235 @@
+  breakdown.feeMultiplier = new CollateralBreakdown(actualFeeMultiplier, "N/A");
+  console.group("** Pre-Fees: **");
+  console.table(breakdown);
+  console.groupEnd();
+
+  /**
+   * @notice RUN THE TEST ONCE AND PRODUCE KNOWN DRIFT
+   */
+  await emp.payFees({ from: sponsor });
+  endingStoreCollateral = await collateral.balanceOf(store.address);
+  actualFeesCollected = endingStoreCollateral.sub(startingStoreCollateral).toString();
+  adjustedCollateral = await emp.totalPositionCollateral();
+  contractCollateral = await collateral.balanceOf(emp.address);
+  rawContractCollateral = await emp.rawTotalPositionCollateral();
+  actualFeeMultiplier = await emp.cumulativeFeeMultiplier();
+  driftTotal = contractCollateral.sub(toBN(adjustedCollateral.rawValue));
+
+  // Test 1) The correct fees are paid are regardless of precision loss.
+  assert.equal(testConfig.expectedFeesCollectedPerPeriod.toString(), actualFeesCollected.toString());
+
+  // Test 2) Due to the precision error mentioned above, `getCollateral()` should return
+  // slightly less than what we are expecting.
+  assert(driftTotal.gt(toBN(0)));
+
+  // Log results.
+  breakdown.expected = new CollateralBreakdown(contractCollateral, "N/A");
+  breakdown.credited = new CollateralBreakdown(adjustedCollateral, "N/A");
+  breakdown.raw = new CollateralBreakdown(rawContractCollateral, "N/A");
+  breakdown.feeMultiplier = new CollateralBreakdown(actualFeeMultiplier, "N/A");
+  breakdown.drift = new CollateralBreakdown(driftTotal, "N/A");
+  console.group(`** After 1 second: ${actualFeesCollected.toString()} collateral collected in fees **`);
+  console.table(breakdown);
+  console.groupEnd();
+
+  /**
+   * @notice RUN THE REMAINDER OF THE TEST AND CHECK UNKNOWN DRIFT
+   * @dev For all of these runs, the expected fees collected per second should still be floor'd to 4 wei.
+   * @dev Since we are no longer dividing by 9,000,000 in an intermediate calculation, it is not obvious that more
+   * rounding errors will occur.
+   */
+  for (let i = 1; i <= testConfig.runs - 1; i++) {
+    await emp.setCurrentTime(startTime.addn(1 + i), { from: contractDeployer });
+    await emp.payFees({ from: sponsor });
+  }
+  endingStoreCollateral = await collateral.balanceOf(store.address);
+  actualFeesCollected = endingStoreCollateral.sub(startingStoreCollateral).toString();
+  adjustedCollateral = await emp.getCollateral(sponsor);
+  contractCollateral = await collateral.balanceOf(emp.address);
+  rawContractCollateral = await emp.rawTotalPositionCollateral();
+  actualFeeMultiplier = await emp.cumulativeFeeMultiplier();
+  driftTotal = contractCollateral.sub(toBN(adjustedCollateral.rawValue));
+
+  // Test 1) The correct fees are paid are regardless of precision loss.
+  assert.equal(
+    (testConfig.expectedFeesCollectedPerPeriod * testConfig.runs).toString(),
+    actualFeesCollected.toString()
+  );
+
+  // Test 2) Let's check if there is more drift.
```

Comment:
No we don't because I still am not confident in the bounds. I thought the max bound would be < `n wei` but your previous comment has me thinking its possible it could blow up like `redeem()`